### PR TITLE
feat: Fixed magic link implementation issues

### PR DESCRIPTION
**Pull request by Homoakin619**
## What did you do
- Generated user profile on user registration
- Fixed issues with authentication endpoints
- Updated test to match updates

### CODE REVIEW

**Review by davidolanrewaju — DISMISSED** [review]

**Review by Cobidex — DISMISSED** [review]

**Review by Heba-WebDev — APPROVED** [review]

**Review by alexindevs — APPROVED** [review]

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -152,32 +162,33 @@ export default class AuthenticationService {
...
             last_name: user.last_name,
             email: user.email,
-            id: user.id,
           },
         },
       };
 
-      return { message: 'Login successful', ...responsePayload };
+      return { message: LOGIN_SUCCESSFUL, ...responsePayload };
     } catch (error) {
```
Comment (Cobidex) [inline src/modules/auth/auth.service.ts:186]: we can do without this try/catch blocks

File: src/modules/otp/otp.service.ts [hunk 2]
```diff
@@ -30,7 +31,7 @@ export class OtpService {
 
       return otp;
     } catch (error) {
-      Logger.error('OtpServiceError ~ createOtpError ~', error);
+      console.log('OtpServiceError ~ createOtpError ~', error);
```
Comment (Cobidex) [inline src/modules/otp/otp.service.ts:34]: the response interceptor is a good point to log these errors, we can do without the try/catch

**Review by Cobidex — APPROVED** [review]
The comments are merely points for our consideration

File: src/modules/otp/otp.service.ts (hunk 2 above)
Comment (Homoakin619) [reply src/modules/otp/otp.service.ts:34]: I created a fix for it in an upcoming pr, Logging it in the interceptor actually does not specify where the error is coming from

File: src/modules/auth/auth.service.ts (hunk 1 above)
Comment (Homoakin619) [reply src/modules/auth/auth.service.ts:186]: we can but it using it here serves a purpose

**Pull request merged**